roomserver/internal/perform/perform_peek.go

```diff
@@ -182,8 +198,6 @@ func (r *Peeker) performPeekRoomByID(
 		}
 	}
 
-	// TODO: handle federated peeks
```

Comment:
I think probably the important thing here is to make sure that we don't attempt a federated peek if someone on the HS is already joined to the room, otherwise we risk rewriting the state unnecessarily. We should be able to trust our own local view of the room if that's the case.